### DISCUSSION TITLE
Used React.ComponentProps for getting FormField props but getting weird ts error in FormField's control prop when passing form.control to control prop

### DISCUSSION THREAD

**Discussion by homocodian** [react-hook-form/react-hook-form#10567] [Q&A] [answered]
Hey guys, I have extracted FormField component of shadcn's ui but I am getting weird error on control prop, used React.componentProps for getting props of FormField component.
Here is the playground
https://stackblitz.com/edit/stackblitz-starters-fkchdm?embed=1&file=pages%2Findex.tsx

**Comment by leapful** [comment, marked as answer]
@Ashish-singh-dev 

You must pass generic field values type to your TextInputProps in order to infer the type of `form.control`.

Working demo: https://stackblitz.com/edit/stackblitz-starters-2acueq?file=pages%2Findex.tsx,components%2Ftext-input.tsx
